lib/IRGen/GenClass.h

```diff
@@ -50,12 +50,10 @@ namespace irgen {
   
   enum class ClassDeallocationKind : unsigned char;
   enum class FieldAccess : uint8_t;
-  
-  OwnedAddress projectPhysicalClassMemberAddress(IRGenFunction &IGF,
-                                                 llvm::Value *base,
-                                                 SILType baseType,
-                                                 SILType fieldType,
-                                                 VarDecl *field);
+
+  OwnedAddress projectPhysicalClassMemberAddress(
+      TypeExpansionContext context, IRGenFunction &IGF, llvm::Value *base,
```

Comment:
If a SILFunction always has a particular TypeExpansionContext, then similarly, maybe IRGenFunction could as well. For `IRGenSILFunction` it could use the context from the function being lowered, and for other IGFs, maybe we could add it as a constructor argument?

Reply:
…though by IRGen type, maybe we can also always use a maximal TypeExpansionContext, since we don't have to worry about cross-module SIL inlining anymore?

Reply:
You are right.
I have IRGenModule::getMaximalTypeExpansionContext() which I should have used here instead.